completions/bash/brew

```diff
@@ -102,8 +102,8 @@ __brew_complete_tapped() {
 __brew_complete_commands() {
   local cur="${COMP_WORDS[COMP_CWORD]}"
   local cmds
-  HOMEBREW_CACHE=$(brew --cache)
-  HOMEBREW_REPOSITORY=$(brew --repo)
+  HOMEBREW_CACHE=${HOMEBREW_CACHE:-$(brew --cache)}
```

Comment:
How about making `brew shellenv` set this? Similarly, is this something the ZSH/Fish completions could benefit from?

Reply:
> How about making `brew shellenv` set this?

I guess `shellenv` can set HOMEBREW_CACHE, but I think this function should work regardless of whether HOMEBREW_CACHE and HOMEBREW_REPOSITORY are set. We can check both of these vars early in the script and bail out with an error message if they're not. I just don't want `brew` to be called twice every time this function is called.

> Similarly, is this something the ZSH/Fish completions could benefit from?

Fish seems to be storing output of `brew --cache` in a variable

https://github.com/Homebrew/brew/blob/3af2f0ececc848bca34ddbd5b876d20dc24a6aaa/completions/fish/brew.fish#L92-L94

and Zsh seems to be caching the final result, so it doesn't need to cache the output of `brew --cache` (perhaps, this approach can be used in Bash completions too).

https://github.com/Homebrew/brew/blob/3af2f0ececc848bca34ddbd5b876d20dc24a6aaa/completions/zsh/_brew#L229-L239

Reply:
> Using `brew shellenv` seems likely to avoid that.

Submitted #14499. I'll update the PR assuming that HOMEBREW_CACHE and HOMEBREW_REPOSITORY are both set.

Reply:
> > perhaps, this approach can be used in Bash completions too
> 
> Yeh, that seems ideal if possible?

If we can just cache this and not have to add to `brew shellenv`: this seems ideal. Let me know when you've ruled this out and can push forward with https://github.com/Homebrew/brew/pull/14499

Reply:
> If we can just cache this and not have to add to `brew shellenv`: this seems ideal. Let me know when you've ruled this out and can push forward with #14499

I've updated the PR and Homebrew commands are now cached into `__HOMEBREW_COMMANDS` env. variable.